packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/ec2/integ.tls-network-load-balanced-ecs-service.ts

```diff
@@ -0,0 +1,55 @@
+import { InstanceType, Vpc } from 'aws-cdk-lib/aws-ec2';
+import { Cluster, ContainerImage } from 'aws-cdk-lib/aws-ecs';
+import { App, Stack } from 'aws-cdk-lib';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import { NetworkLoadBalancedEc2Service } from 'aws-cdk-lib/aws-ecs-patterns';
+import { Certificate, CertificateValidation } from 'aws-cdk-lib/aws-certificatemanager';
+import { PublicHostedZone } from 'aws-cdk-lib/aws-route53';
+
+/**
+ * In order to test this you need
+ * to have a valid public hosted zone that you can use
+ * to request certificates for.
+ *
+*/
+const hostedZoneId = process.env.CDK_INTEG_HOSTED_ZONE_ID ?? process.env.HOSTED_ZONE_ID;
+if (!hostedZoneId) throw new Error('For this test you must provide your own HostedZoneId as an env var "HOSTED_ZONE_ID". See framework-integ/README.md for details.');
+const hostedZoneName = process.env.CDK_INTEG_HOSTED_ZONE_NAME ?? process.env.HOSTED_ZONE_NAME;
+if (!hostedZoneName) throw new Error('For this test you must provide your own HostedZoneName as an env var "HOSTED_ZONE_NAME". See framework-integ/README.md for details.');
+const domainName = process.env.CDK_INTEG_DOMAIN_NAME ?? process.env.DOMAIN_NAME;
+if (!domainName) throw new Error('For this test you must provide your own DomainName as an env var "DOMAIN_NAME". See framework-integ/README.md for details.');
+
+const app = new App();
+const stack = new Stack(app, 'tls-network-load-balanced-ecs-service');
+const vpc = new Vpc(stack, 'Vpc', { maxAzs: 2 });
+const cluster = new Cluster(stack, 'Cluster', { vpc });
+
+cluster.addCapacity('DefaultAutoScalingGroupCapacity', {
+  instanceType: new InstanceType('t2.small'),
+  desiredCapacity: 2,
+});
+
+const hostedZone = PublicHostedZone.fromHostedZoneAttributes(stack, 'HostedZone', {
+  hostedZoneId,
+  zoneName: hostedZoneName,
+});
+const validation = CertificateValidation.fromDns(hostedZone);
+
+// EC2 Service and NLB with TLS listener
+new NetworkLoadBalancedEc2Service(stack, 'myServiceWithTls', {
+  cluster,
+  memoryLimitMiB: 256,
+  listenerCertificate: new Certificate(stack, 'myCert', {
+    domainName,
+    validation,
+  }),
```

Comment:
since creating the certificate is not important for this testing, could you please modify the integration for EC2, and Fargate to accept an imported certificate, as the certificate creation step takes a very long time, and slow down this testing.

Reply:
Sounds good to me.

To confirm I understand your suggestion: 
I can use 
https://github.com/aws/aws-cdk/blob/ab73e538b7b77df8f8e981de2152dde814f66a1d/packages/%40aws-cdk-testing/framework-integ/test/aws-apigateway/test/integ.domain-name.ts#L13
and then import a certificate with 
https://github.com/aws/aws-cdk/blob/ab73e538b7b77df8f8e981de2152dde814f66a1d/packages/%40aws-cdk-testing/framework-integ/test/aws-apigateway/test/integ.domain-name.ts#L83

Reply:
and for the certificate creation itself, I believe you can create a self signed certificate and then import it, I believe this will be an easier solution.

Reply:
you can follow the following steps to create a self signed certificate:
```
$ openssl genrsa -out my-private.key 2048
$ openssl req -new -key my-private.key -out my-csr.csr
$ openssl x509 -req -days 365 -in my-csr.csr -signkey my-private.key -out my-certificate.crt
```

Then you can follow [this link](https://docs.aws.amazon.com/acm/latest/userguide/import-certificate-api-cli.html#import-certificate-api) to import it.

Also, please add these steps to the test case

Reply:
Thanks for the guidance!  

I will take me some time to learn the details of self signed certificate and update the tests.

Reply:
Hi @moelasmar,  

I tested my change with Amazon issued cert at the end.  
And the update code works. 

---

When I use the self signed cert, all AWS resources exception the NLB created successfully. The NLB failed with the following message in CloudFormation deployment events:  
`Resource handler returned message: "The certificate 'arn:aws:acm:us-east-1:128346755879:certificate/efe2138c-cf1d-481b-ba56-737b1ca1819f' must have a fully-qualified domain name, a supported signature, and a supported key size. (Service: ElasticLoadBalancingV2, Status Code: 400, Request ID: 49557e15-b32d-43f9-bb26-7ef309de21a7)" (RequestToken: 5c8f3642-cc69-6051-e9d7-a4b6fdc41064, HandlerErrorCode: InvalidRequest)`